framework/integration/test/reconciliation/reconciliation_test.go

```diff
@@ -1,21 +1,113 @@
-// +build k8srequired
-
 package reconciliation
 
 import (
+	"reflect"
 	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	"github.com/giantswarm/operatorkit/framework/integration/client"
 )
 
 // Test_Finalizer_Integration_Reconciliation is a integration test for
 // the proper replay and reconciliation of delete events with finalizers.
 func Test_Finalizer_Integration_Reconciliation(t *testing.T) {
+	configMapName := "test-cm"
+	testFinalizer := "operatorkit.giantswarm.io/test-operator"
 	testNamespace := "finalizer-integration-reconciliation-test"
+	testOtherFinalizer := "operatorkit.giantswarm.io/other-operator"
+	operatorName := "test-operator"
 
 	client.MustSetup(testNamespace)
 	defer client.MustTeardown(testNamespace)
 
-	// TODO: Implement the actual test here.
+	operatorkitFramework, err := client.NewFramework(operatorName, testNamespace)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We create a configmap, but add a finalizer of another operator. This will
+	// cause the configmap to continue existing after the framework removes it own
+	// finalizer.
+	cm := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: testNamespace,
+			Finalizers: []string{
+				testOtherFinalizer,
+			},
+		},
+		Data: map[string]string{},
+	}
+	_, err = client.CreateConfigMap(cm)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	// We start the framework.
+	go operatorkitFramework.Boot()
+
+	// We wait the absolute maximum amount of time here:
+	// 20 second ResyncPeriod + 2 second RateWait + 3 second for safety.
+	time.Sleep(25 * time.Second)
```

Comment:
Really dumb right now, getting information from the informer would be the easiest way to do this. I also thought about using `backoff` and polling the k8s api.

Reply:
You can write an informer wrapper which will do that for you. There is an `informer.Interface`.

Reply:
I will do this now.

Reply:
I went through this a bit, unsure if an informer wrapper will benefit this test much. It will add even more complexity and we can verify the desired results with the resource now.

WDYT @kopiczko

Reply:
What information are missing? You get all info from your test resource. I do not see the need to wrap the informer again.

Reply:
The top comment is before the resource, I don't think anything is missing. Just want to make sure Mr. Pawel is happy :D

Reply:
I think it's needed to know when events were emitted. Still we don't have guarantee they are processed.

This should be good enough. To shorten the execution time you can use BackOff. But I'm not fussed about that ATM. Can be improved later if needed.